libPhoneNumber-GeocodingParser/libPhoneNumber-GeocodingParser/NBGeocoderMetadataParser.m

```diff
@@ -0,0 +1,74 @@
+//
+//  NBGeocoderMetadataParser.m
+//  libPhoneNumber-GeocodingParser
+//
+//  Created by Rastaar Haghi on 7/1/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import "NBGeocoderMetadataParser.h"
+
+@implementation NBGeocoderMetadataParser {
+@private
+  SQLiteDatabaseConnection *_databaseConnection;
+}
+
+- (instancetype)init {
+  self = [super init];
+  return self;
+}
+
+- (void)convertFileToSQLiteDatabase:(NSString *)completeTextFilePath
+                       withFileName:(NSString *)textFileName
+                       withLanguage:(NSString *)languageCode {
+  NSString *fileName = completeTextFilePath;
+  NSData *data = [NSData dataWithContentsOfFile:fileName];
+  NSString *myStr = [[NSString alloc] initWithData:data
+                                          encoding:NSUTF8StringEncoding];
+  NSMutableArray<NSString *> *countryCodes =
+      (NSMutableArray *)[textFileName componentsSeparatedByString:@"."];
```

Comment:
Casting here is incorrect. The API returns non-mutable NSArray object, to make it mutable, you call `mutableCopy` on the array to get NSMutableArray back.

Reply:
never needed to be NSMutableArray, just left it as NSArray.